src/Routing/Route/DashedRoute.php

```diff
@@ -56,11 +56,12 @@ protected function _camelizePlugin($plugin)
      * camelBacked form.
      *
      * @param string $url The URL to parse
+     * @param string $method The HTTP method.
      * @return array|false An array of request parameters, or false on failure.
      */
-    public function parse($url)
+    public function parse($url, $method = '')
```

Comment:
> Or we can just not care and tell people to silence E_STRICT during the migration time

Doing so would account for `AuthComponent::setUser()` and `StorageInterface::write()` typehint changes too :)